homeassistant/components/otbr/__init__.py

```diff
@@ -0,0 +1,222 @@
+        raise HomeAssistantError
+
+    try:
+        return OperationalDataSet.from_json(await response.json())
+    except (JSONDecodeError, vol.Error) as exc:
+        raise HomeAssistantError from exc
+
+
+async def async_get_active_dataset_tlvs(hass: HomeAssistant) -> bytes | None:
+    """Get current active operational dataset in TLVS format, or None.
+
+    Returns None if there is no active operational dataset.
+    Raises if the http status is 400 or higher or if the response is invalid.
+    """
+
+    response = await async_get_clientsession(hass).get(
+        f"{await _async_get_thread_rest_service_url(hass)}/node/dataset/active",
+        headers={"Accept": "text/plain"},
+        timeout=aiohttp.ClientTimeout(total=10),
+    )
+
+    _raise_for_status(response)
+    if response.status == HTTPStatus.NO_CONTENT:
+        return None
+
+    if response.status != HTTPStatus.OK:
+        raise HomeAssistantError
+
+    try:
+        tmp = await response.read()
+        return bytes.fromhex(tmp.decode("ASCII"))
+    except ValueError as exc:
+        raise HomeAssistantError from exc
+
+
+async def async_create_active_dataset(
+    hass: HomeAssistant, dataset: OperationalDataSet
+) -> None:
+    """Create active operational dataset.
+
+    The passed in OperationalDataSet does not need to be fully populated, any fields
+    not set will be automatically set by the open thread border router.
+    Raises if the http status is 400 or higher or if the response is invalid.
+    """
+
+    response = await async_get_clientsession(hass).post(
+        f"{await _async_get_thread_rest_service_url(hass)}/node/dataset/active",
+        json=dataset.as_json(),
+        timeout=aiohttp.ClientTimeout(total=10),
+    )
+
+    if response.status == HTTPStatus.CONFLICT:
+        raise ThreadNetworkActiveError
+    _raise_for_status(response)
+    if response.status != HTTPStatus.ACCEPTED:
+        raise HomeAssistantError
+
+
+async def async_set_active_dataset(
+    hass: HomeAssistant, dataset: OperationalDataSet
+) -> None:
+    """Set current active operational dataset.
+
+    The passed in OperationalDataSet does not need to be fully populated, any fields
+    not set will be automatically copied from the current active dataset.
+    Raises if the http status is 400 or higher or if the response is invalid.
+    """
+
+    response = await async_get_clientsession(hass).put(
+        f"{await _async_get_thread_rest_service_url(hass)}/node/dataset/active",
+        json=dataset.as_json(),
+        timeout=aiohttp.ClientTimeout(total=10),
+    )
+
+    if response.status == HTTPStatus.NOT_FOUND:
+        raise NoDatasetError
+    if response.status == HTTPStatus.CONFLICT:
+        raise ThreadNetworkActiveError
+    _raise_for_status(response)
+    if response.status != HTTPStatus.ACCEPTED:
+        raise HomeAssistantError
+
+
+async def async_set_active_dataset_tlvs(hass: HomeAssistant, dataset: bytes) -> None:
```

Comment:
Why are we exposing all these functions. It looks like we allow them to be accessed by other integrations ?

Reply:
Yes, that's the point of the integration?